### PR TITLE
Update auto control

### PR DESCRIPTION
**Pull request by InvincibleRMC**
# Pull Request

## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Updates to match picture in discord.

## Related Tickets & Documents

- Related Issue #
#95

### CODE REVIEW

File: src/surface/flight_control/flight_control/control_inverter_node.py [hunk 1]
```diff
@@ -48,67 +23,37 @@ def __init__(self) -> None:
...
-        rc_msg.channels[THROTTLE_CHANNEL] = msg.vertical
-        rc_msg.channels[LATERAL_CHANNEL] = msg.lateral
-        rc_msg.channels[PITCH_CHANNEL] = msg.pitch
-        rc_msg.channels[YAW_CHANNEL] = msg.yaw
-        rc_msg.channels[ROLL_CHANNEL] = msg.roll
-
-        return rc_msg
-
-    def invert_callback(self, msg: CameraControllerSwitch) -> None:
+    def invert_callback(self, _: CameraControllerSwitch) -> None:
```
Comment (benjaminwp18) [inline src/surface/flight_control/flight_control/control_inverter_node.py:46]:
```suggestion
    def invert_callback(self, _msg: CameraControllerSwitch) -> None:
```

File: src/surface/gui/gui/widgets/task_selector.py [hunk 2]
```diff
@@ -10,68 +12,42 @@
...
         self.setLayout(layout)
 
         # Create Start button
-        self.start_btn = QPushButton("Start Auto Docking")
-        self.start_btn.clicked.connect(self.start_btn_clicked)
+        self.start_btn = QPushButton("Start Auto")
+        self.start_btn.clicked.connect(partial(self.task_controller.send_request_async,
+                                       AutonomousFlight.Request(
+                                           state=AutonomousFlight.Request.START
+                                           )))
```
Comment (benjaminwp18) [inline src/surface/gui/gui/widgets/task_selector.py:29]:
```python
self.start_btn.clicked.connect(
    lambda: self.task_controller.send_request_async(
        AutonomousFlight.Request(state=AutonomousFlight.Request.START)
    )
)
```

File: src/surface/gui/gui/widgets/task_selector.py [hunk 3]
```diff
@@ -10,68 +12,42 @@
...
         self.start_btn.setFixedWidth(WIDTH)
 
         # Create Stop button
-        self.stop_btn = QPushButton("Stop Auto Docking")
-        self.stop_btn.clicked.connect(self.stop_btn_clicked)
+        self.stop_btn = QPushButton("Stop Auto")
+        self.stop_btn.clicked.connect(partial(self.task_controller.send_request_async,
+                                      AutonomousFlight.Request(
+                                           state=AutonomousFlight.Request.STOP
+                                           )))
```
Comment (benjaminwp18) [inline src/surface/gui/gui/widgets/task_selector.py:38]: See above

File: src/surface/gui/gui/widgets/task_selector.py [hunk 4]
```diff
@@ -10,68 +12,42 @@
...
-            'GUI changed task to: Manual Control')
-
-        self.task_status.setText('Auto Docking: Disabled')
-
-        self.task_controller.send_request_async(
-            AutonomousFlight.Request(start=False))
 
     @pyqtSlot(AutonomousFlight.Response)
-    def handle_scheduler_response(self, response: AutonomousFlight.Response) -> None:
+    def handle_scheduler_response(self, _: AutonomousFlight.Response) -> None:
```
Comment (benjaminwp18) [inline src/surface/gui/gui/widgets/task_selector.py:49]:
```suggestion
    def handle_scheduler_response(self, _response: AutonomousFlight.Response) -> None:
```

File: src/surface/gui/gui/widgets/task_selector.py [hunk 5]
```diff
@@ -10,68 +12,42 @@
...
         """Handle scheduler response to request sent from gui_changed_task."""
-        msg = 'Auto docking is '
-        if response.is_running:
-            msg += 'now running'
-        else:
-            msg += 'no longer running'
-        self.task_controller.get_logger().info(msg)
```
Comment (benjaminwp18) [inline src/surface/gui/gui/widgets/task_selector.py:77]: Somehow show state change on GUI when response is received

**Review by benjaminwp18 — CHANGES_REQUESTED** [review]

File: src/surface/flight_control/flight_control/auto_docking_node.py [hunk 6]
```diff
@@ -1,28 +1,34 @@
...
         self.running = False
 
         # TODO: Add cam frame subscriber here to act as control loop for auto docking
 
     def task_control_callback(self, request: AutonomousFlight.Request,
                               response: AutonomousFlight.Response) -> AutonomousFlight.Response:
-        self.running = request.start
-        response.is_running = self.running
-
+        self.running = request.state
```
Comment (benjaminwp18) [inline src/surface/flight_control/flight_control/auto_docking_node.py:30]:
```suggestion
        self.current_state = request.state
```

File: src/surface/flight_control/flight_control/auto_docking_node.py [hunk 7]
```diff
@@ -1,28 +1,34 @@
...
-            AutonomousFlight, 'auto_docker_control', self.task_control_callback)
+            AutonomousFlight, 'auto_control_toggle', self.task_control_callback)
+
+        self.pixhawk_control = self.create_publisher(
+            PixhawkInstruction,
+            "pixhawk_control",
+            QoSPresetProfiles.DEFAULT.value,
+        )
 
         self.running = False
```
Comment (benjaminwp18) [inline src/surface/flight_control/flight_control/auto_docking_node.py:24]:
```suggestion
        self.current_state = AutonomousFlight.Request.STOP
```

**Review by benjaminwp18 — CHANGES_REQUESTED** [review]

**Review by benjaminwp18 — APPROVED** [review]
lgtm

**Comment by NoahMollerstuen** [comment]
Can you merge this with auto transplanting instead?

**Pull request merged**